### DISCUSSION TITLE
Zeppelin contracts library installation

### DISCUSSION THREAD

**Discussion by taofeeq125** [Cyfrin/security-and-auditing-full-course-s23#83] [Q&A]
My terminal keeps bringing out this error below, what can I do:

git submodule exited with code 128

**Comment by johnsonstephan** [comment]
Could be an issue with Git config, if so - check here: https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration

Else, you may find value in exploring solutions with AI assistance (like GPT) to discover the exact issue / resolution.

**Reply by taofeeq125** [reply]
I have tried it for days now, I didn't get the solution to the error sir.

**Reply by johnsonstephan** [reply]
At this point, if it's hindering your progress with the course, a clean install may be best. You can also opt to setup a virtual machine to ensure you have a clean slate for setup.

**Reply by taofeeq125** [reply]
How can it be done sir?

**Comment by farismln** [comment]
I have this error too a couple of days ago because of bad internet connection. Maybe you should check is there any apps that using your internet bandwidth?

**Comment by gaborbalazseng** [comment]
I have this error too.

`fatal: No url found for submodule path 'lib/forge-std' in .gitmodules Error:  ←[31mgit submodule exited with code 128←[0m make: *** [Makefile:15: install] Error 1`

When running make, the command replaces the forward slashes (/) with two back slashes (\\) in the submodule and path values.

**Comment by akhilmanga** [comment]
Replace your make file with this:

```
-include .env

.PHONY: all test clean deploy fund help install snapshot format anvil scopefile

DEFAULT_ANVIL_KEY := 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80

all: remove install build

# Clean the repo
clean  :; forge clean

# Remove modules
remove :; rm -rf .gitmodules && rm -rf .git/modules/* && rm -rf lib && touch .gitmodules && git add . && git commit -m "modules"

install :; forge install foundry-rs/forge-std --no-commit && forge install openzeppelin/openzeppelin-contracts --no-commit && forge install openzeppelin/openzeppelin-contracts-upgradeable --no-commit 

# Update Dependencies
update:; forge update

build:; forge build

test :; forge test 

snapshot :; forge snapshot

format :; forge fmt

anvil :; anvil -m 'test test test test test test test test test test test junk' --steps-tracing --block-time 1

slither :; slither . --config-file slither.config.json --checklist 

scope :; tree ./src/ | sed 's/└/#/g; s/──/--/g; s/├/#/g; s/│ /|/g; s/│/|/g'

scopefile :; @tree ./src/ | sed 's/└/#/g' | awk -F '── ' '!/\.sol$$/ { path[int((length($$0) - length($$2))/2)] = $$2; next } { p = "src"; for(i=2; i<=int((length($$0) - length($$2))/2); i++) if (path[i] != "") p = p "/" path[i]; print p "/" $$2; }' > scope.txt

aderyn :; aderyn .

```

Run `make` command in the terminal. Openzeppelin contracts will be installed.

**Reply by cromewar** [reply]
@taofeeq125 does this work for you?

**Reply by Shawon-Mandal** [reply]
@akhilmanga this still shows the same error

**Comment by theirrationalone** [comment]
1. just `copy-paste` the patrick's `.gitmodules` file (relative to your clone's origin) in your project. (Vs-Code Restart required)
2. follow-up the triage suggested by @akhilmanga . Vs-Code Restart required)
3. delete the file -> make it again -> hardcode the repos path -> save it -> restart vs code -> worked ? ✔: add remapping 
4. copy paste all of your work in a safe place -> delete the project -> re-clone it 😅

maybe voila ;`)

**Reply by cromewar** [reply]
Does this work for you @taofeeq125??
